backend/app/api/endpoints/adapter/shells.py

```
@@ -0,0 +1,741 @@
+            validation_id=validation_id,
+            status=request.status,
+            stage=request.stage,
+            progress=request.progress,
+            valid=request.valid,
+            checks=[c.model_dump() for c in request.checks] if request.checks else None,
+            errors=request.errors,
+            error_message=request.errorMessage,
+        )
+
+        if not success:
+            raise HTTPException(status_code=500, detail="Failed to update validation status")
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)